src/app/admin/admin-products/admin-products.component.ts

```diff
@@ -1,7 +1,33 @@
-import { Component } from '@angular/core';
+import { Component, OnDestroy } from '@angular/core';
+import { ProductService } from 'src/app/product.service';
+import { Observable, Subscription } from 'rxjs';
+import { Product } from 'src/app/models/product';
 
 @Component({
   selector: 'app-admin-products',
   templateUrl: './admin-products.component.html'
 })
-export class AdminProductsComponent {}
+export class AdminProductsComponent implements OnDestroy {
+  productsObservable = [];
+  filteredProducts: any[];
```

Comment:
we should create a model for product. using any defeats the point of using typescript. Generally you wanna avoid using any unless you have to